code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -200,7 +200,7 @@ public Account createAccount(String name, String kind, String externalId, String
     }
 
     @Override
-    public Identity getIdentity(Long id) {
+    public Identity getIdentity(Long id, boolean obfuscate) {
```

Comment:
I think this needs to be done in a slightly different fashion. When obfuscate is passed in, the code implies that the ApiContext might not be null at this point (line 213). Just pass in IdFormatter to the call; and if its null - return the id non-obfuscated.